packages/edit-site/src/components/editor/index.js

```diff
@@ -125,25 +129,31 @@ function Editor() {
 	// and Query Pagination blocks.
 	const blockContext = useMemo(
 		() => ( {
-			...page.context,
-			query: page.context.query || { categoryIds: [] },
+			...page?.context,
+			query: page?.context.query || { categoryIds: [] },
 			queryContext: [
-				page.context.queryContext || { page: 1 },
+				page?.context.queryContext || { page: 1 },
 				( newQueryContext ) =>
 					setPage( {
 						...page,
 						context: {
-							...page.context,
+							...page?.context,
 							queryContext: {
-								...page.context.queryContext,
+								...page?.context.queryContext,
 								...newQueryContext,
 							},
 						},
 					} ),
 			],
 		} ),
-		[ page.context ]
+		[ page?.context ]
 	);
+
+	let entityId;
+	if ( templateType ) {
+		entityId = templateType === 'wp_template' ? templateId : templatePartId;
+	}
+
```

Comment:
Maybe move this before R85 so we can use `entityId` on [R89-R91](https://github.com/WordPress/gutenberg/pull/23775/files#diff-641afe1a0152b0e5136c8b46edb84b33R89-R91)?

Reply:
Hm, maybe we can add an entityId to the select callback and then it could be available there

Reply:
That makes sense to me.

Since `entityId` is derived from values that are available in `useSelect` anyway, we could consolidate the logic in there.